laws/src/main/scala/cats/laws/DecideableLaws.scala

```diff
@@ -0,0 +1,28 @@
+package cats
+package laws
+
+trait DecideableLaws[F[_]] extends ContravariantMonoidalLaws[F] {
+  implicit override def F: Decideable[F]
+
+  def decideableDecideRightAbsorption[A](fa: F[A]): IsEq[F[A]] =
+    (F.decide(F.trivial[A], fa))(Right.apply[A, A]) <-> fa
+
+  def decideableSumAssociativity[A, B, C](fa: F[A],
+                                          fb: F[B],
+                                          fc: F[C]): (F[Either[A, Either[B, C]]], F[Either[Either[A, B], C]]) =
+    (F.sum(fa, F.sum(fb, fc)), F.sum(F.sum(fa, fb), fc))
+
+  def decideableRightDistributivity[A, B, C](fa: F[A], fb: F[B], f: C => A, g: C => B): IsEq[F[Either[C, C]]] =
```

Comment:
This law tests for distributivity of `sum` and `contramap`, but I think we should also have one for `sum` and `product` :)

Reply:
Oooo good idea. I'll add that.